File: Solutions/0169/0169.java
// Author: washing
// DateTime: 2021/1/26 18:52
// File: 0169.java
// Desc: 

class Solution {
    public int majorityElement(int[] nums) {
        int a = nums[0];
        int c = 1;
        int i;
        for (int n=1; n<nums.length;n++){
            i = nums[n];
            if (c == 0){
                a = i;
                c++;
            }else if (i == a){
                c++;
            }else{
                c--;
            }
        }
        return a;
    }
}

File: Solutions/0169/0169.cpp
// Author: washing
// DateTime: 2021/1/26 18:50
// File: 0169.cpp
// Desc: 


class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int a=0;
        int c=0;
        for (auto i : nums){
            if (c==0){
                a = i;
                c++;
            }else if (a==i){
                c++;
            }else {
                c--;
            }
        }
        return a;
    }
};